packages/logger/src/app-insights-logger-client.spec.ts

```diff
@@ -199,6 +201,97 @@ describe(AppInsightsLoggerClient, () => {
+            setupCallsForTelemetrySetup();
+            await testSubject.setup(null);
+            appInsightsTelemetryClientMock.reset();
+
+            const customDimensionAssignments = { scanId: 'scan-id', batchRequestId: 'batch-req-id' };
+            appInsightsTelemetryClientMock
+                .setup(t =>
+                    t.trackTrace(
+                        It.isValue({
+                            message: 'trace1',
+                            severity: appInsights.Contracts.SeverityLevel.Information,
+                            properties: customDimensionAssignments,
+                        }),
+                    ),
+                )
+                .verifiable();
+            testSubject.setCustomProperties(customDimensionAssignments);
+            testSubject.log('trace1', LogLevel.info);
+            verifyMocks();
+        });
+
+        it('set custom dimensions for trackMetric', async () => {
+            setupCallsForTelemetrySetup();
+            await testSubject.setup(null);
+            appInsightsTelemetryClientMock.reset();
+
+            const customDimensionAssignments = { scanId: 'scan-id', batchRequestId: 'batch-req-id' };
+            appInsightsTelemetryClientMock
+                .setup(t =>
+                    t.trackMetric(
+                        It.isValue({
+                            name: 'metric1',
+                            value: 10,
+                            properties: customDimensionAssignments,
+                        }),
+                    ),
+                )
+                .verifiable();
+            testSubject.setCustomProperties(customDimensionAssignments);
+            testSubject.trackMetric('metric1', 10);
+            verifyMocks();
+        });
+
+        it('set custom dimensions for trackException', async () => {
+            setupCallsForTelemetrySetup();
+            await testSubject.setup(null);
+            appInsightsTelemetryClientMock.reset();
+
+            const error = new Error('some error');
+            const customDimensionAssignments = { scanId: 'scan-id', batchRequestId: 'batch-req-id' };
+            appInsightsTelemetryClientMock
+                .setup(t =>
+                    t.trackException(
+                        It.isValue({
+                            exception: error,
+                            properties: customDimensionAssignments,
+                        }),
+                    ),
+                )
+                .verifiable();
+            testSubject.setCustomProperties(customDimensionAssignments);
+            testSubject.trackException(error);
+            verifyMocks();
+        });
+
+        it('set custom dimensions for trackEvent', async () => {
+            setupCallsForTelemetrySetup();
+            await testSubject.setup(null);
+            appInsightsTelemetryClientMock.reset();
+
+            const customDimensionAssignments = { scanId: 'scan-id', batchRequestId: 'batch-req-id' };
```

Comment:
NIT: Can we add a test case in which one of the preset custom properties is overridden by the data passed into ``trackEvent()``?